rasa/utils/tensorflow/model_data.py

```diff
@@ -458,21 +531,35 @@ def _get_shapes_types(self) -> Tuple:
         types = []
         shapes = []
 
-        def append_shape(features: np.ndarray) -> None:
-            if isinstance(features[0], scipy.sparse.spmatrix):
-                # scipy matrix is converted into indices, data, shape
-                shapes.append((None, features[0].ndim + 1))
-                shapes.append((None,))
-                shapes.append((features[0].ndim + 1))
-            elif features[0].ndim == 0:
-                shapes.append((None,))
-            elif features[0].ndim == 1:
-                shapes.append((None, features[0].shape[-1]))
+        def append_shape(_features: FeatureArray) -> None:
+            if _features.number_of_dimensions == 4:
```

Comment:
is there a reason why `number_of_dimensions == 4` is first? Can it be put as one of the elifs below?